src/dbnode/storage/repair_test.go

```diff
@@ -675,11 +695,131 @@ func TestDatabaseRepairPrioritizationLogic(t *testing.T) {
 			ns1.EXPECT().ID().Return(ident.StringID("ns1")).AnyTimes()
 			ns2.EXPECT().ID().Return(ident.StringID("ns2")).AnyTimes()
 
-			ns1.EXPECT().Repair(gomock.Any(), tc.expectedNS1Repair.repairRange)
-			ns2.EXPECT().Repair(gomock.Any(), tc.expectedNS2Repair.repairRange)
+			ns1.EXPECT().Repair(gomock.Any(), tc.expectedNS1Repair.expectedRepairRange)
+			ns2.EXPECT().Repair(gomock.Any(), tc.expectedNS2Repair.expectedRepairRange)
 
 			mockDatabase.EXPECT().OwnedNamespaces().Return(namespaces, nil)
 			require.Nil(t, repairer.Repair())
 		})
 	}
 }
+
+// Database repairer repairs blocks in decreasing time ranges for each namespace. If database repairer fails to
+// repair a time range of a namespace then instead of skipping repair of all past time ranges of that namespace, test
+// that database repaier tries to repair the past corrupt time range of that namespace.
+func TestDatabaseRepairSkipsPoisonShard(t *testing.T) {
+	ctrl := gomock.NewController(t)
```

Comment:
Create the controller inside the subtest, otherwise it will be shared between all the tests, leading to weird outcomes with the expectations.

Reply:
I copied this pattern from the previous test in the same file. Let me know if this needs to be fixed in all the tests in this file.